src/SharedUtilities/models.jl

```diff
@@ -29,13 +32,49 @@ Returns the prognostic variable symbols for the model in the form of a tuple.
 """
 prognostic_vars(m::AbstractModel) = ()
 
+"""
+   prognostic_types(m::AbstractModel{FT}) where {FT}
+
+Returns the prognostic variable types for the model in the form of a tuple.
+
+Types provided must have `zero(T::DataType)` defined. Common examples
+ include
+- Float64, Float32 for scalar variables (a scalar value at each 
+coordinate point)
+- NTuple{k,Float64} for a k-length mutable but statically sized tuple of
+ length `k` at each coordinate point.
+
+Here, the coordinate points are those returned by coordinates(model).
+"""
+prognostic_types(m::AbstractModel) = ()
 """
    auxiliary_vars(m::AbstractModel)
 
 Returns the auxiliary variable symbols for the model in the form of a tuple.
 """
 auxiliary_vars(m::AbstractModel) = ()
 
+
+"""
+   auxiliary_types(m::AbstractModel{FT}) where {FT}
+
+Returns the auxiliary variable types for the model in the form of a tuple.
+
+Types provided must have `zero(T::DataType)` defined. Common examples
+ include
+- Float64, Float32 for scalar variables (a scalar value at each 
+coordinate point)
+- NTuple{k,Float64} for a tuple of
+ length `k` at each coordinate point.
+- Note that Arrays, MVectors are not isbits and cannot be used.
+
+Here, the coordinate points are those returned by coordinates(model).
+"""
+auxiliary_types(m::AbstractModel) = ()
+
+Base.zero(T::DataType = NTuple{k <: Int, FT <: AbstractFloat}) =
+    ntuple(i -> T.types[1](0.0), length(T.types))
```

Comment:
This is probably not a good idea (overloading a Julia Base method for Julia Base types)

Reply:
oh geez, for some reason when I tried `zero(NTuple{N,FT})` (with N and FT known) it didnt work, but it clearly works now that I try it again.

thanks for pointing this out.